src/app/plugins/pubsub.plugin.ts

```diff
@@ -1,23 +1,116 @@
+import Constants from '../../common/constants';
 import { Plugin } from '../../common/plugin';
-import { IContainer, IMessage, ChannelType } from '../../common/types';
+import { ChannelType, IContainer, IHttpResponse, IMessage } from '../../common/types';
+import { RichEmbed } from 'discord.js';
 
 export class PubSubPlugin extends Plugin {
   public name: string = 'Pub Sub Plugin';
-  public description: string = 'Are publix chicken tender subs on sale?';
-  public usage: string = '';
-  public pluginAlias = [];
-  public permission: ChannelType = ChannelType.Bot;
+  public description: string =
+    'Get them prices for your subs, and a steamy picture of the subs you need in your life.';
+  public usage: string = 'pubSub <sub (optional)>';
+  public pluginAlias = ['sub', 'subs', 'sandwich', 'samwich', 'sarnie'];
+  public permission: ChannelType = ChannelType.Public;
+  public pluginChannelName: string = Constants.Channels.Public.Food;
+
+  private _API_URL: string = 'https://pubsub-api.dev/';
+  private _subs: string[] = [];
+  private _embedSubs = new RichEmbed();
 
   constructor(public container: IContainer) {
     super();
+    this.updateList();
+  }
+
+  private updateList() {
+    this.container.httpService
+      .get(`${this._API_URL}allsubs/`)
+      .then((response: IHttpResponse) => {
+        const subs = response.data;
+
+        this._subs = subs.map((subData: { name: string }) => {
+          return subData.name
+            .toLowerCase()
+            .split('-')
+            .join(' ');
+        });
+        this._generateEmbedSubs();
+      })
+      .catch((err) => this.container.loggerService.warn(err));
   }
 
   public async execute(message: IMessage, args?: string[]) {
-    const state = this.container.storeService.get('pubSubStore').state;
-    if (state?.onSale) {
-      message.reply('Chicken tender subs are on sale :eyes:');
-    } else {
-      message.reply('Chicken tender subs are **NOT** on sale.');
+    if (args === undefined || args.length === 0) {
+      args = [''];
+    }
+
+    if (args[0].includes('type') || args[0].includes('list')) {
+      //Simply return the list of supported subs
+      message.reply(this._embedSubs);
+      return;
+    }
+
+    const subIn = this._parseCommand(args);
+
+    // checks if their sub was a sub, then if that sub is recognised
+    const subEntry = this._subs.find((sub: string) => sub === subIn) || '';
```

Comment:
you don't need to cast this to be an empty string. this returns undefined and it is good enough for your check below

Reply:
I actually do since when subEntry === '' that get Abes API to return a random sub.  The other route would be to change the const to a var and reassign it upon reading.  But I personly think that would be counterproductive.